### PR TITLE
Correct the multinode training doc

### PR DESCRIPTION
**Pull request by vanbasten23**
This PR
- adds `--pjrt_distributed` flag back due to https://github.com/pytorch/xla/pull/5732#issuecomment-1783549211.
- makes some correction to the multinode training doc.
- fix remaining comments in the previous pr https://github.com/pytorch/xla/pull/5704

### CODE REVIEW

File: docs/pjrt.md [hunk 1]
```diff
@@ -200,7 +200,7 @@ To use GPUs with PJRT, simply set `PJRT_DEVICE=GPU` and configure
 `GPU_NUM_DEVICES` to the number of devices on the host. For example:
 
 ```
-PJRT_DEVICE=GPU GPU_NUM_DEVICES=4 python3 xla/test/test_train_mp_imagenet.py --fake_data --pjrt_distributed --batch_size=128 --num_epochs=1
+PJRT_DEVICE=GPU GPU_NUM_DEVICES=4 python3 xla/test/test_train_mp_imagenet.py --fake_data --batch_size=128 --num_epochs=1
 ```
 
 You can also use `torchrun` to initiate the single-node multi-GPU training. For example,
```
Comment (jonb377) [inline docs/pjrt.md:206]: Could you please also correct the `test_train_mp_imagenet_torchrun` down below?

Comment (vanbasten23) [reply docs/pjrt.md:206]: oops, good catch!

**Review by will-cromar — APPROVED** [review]

File: test/test_train_mp_mnist.py [hunk 2]
```diff
@@ -73,7 +76,7 @@ def _train_update(device, step, loss, tracker, epoch, writer):
 
 
 def train_mnist(flags, **kwargs):
-  if flags.ddp:
+  if flags.ddp or flags.pjrt_distributed:
```
Comment (jonb377) [inline test/test_train_mp_mnist.py:79]: Ah interesting... Do we need a dedicated flag still, or can we just also check for torchrun some other way? I saw [dist.is_torchelastic_launched](https://pytorch.org/docs/stable/distributed.html) elsewhere in the codebase: https://github.com/pytorch/xla/blob/83778f0/torch_xla/_internal/rendezvous.py#L20

**Review by jonb377 — APPROVED** [review]
LGTM, nothing blocking. Thanks Xiongfei!

**Comment by vanbasten23** [comment]
Thanks for the review!

**Pull request merged**